spdx-utils/src/main/kotlin/SpdxExpression.kt

```diff
@@ -236,6 +253,31 @@ class SpdxCompoundExpression(
             SpdxOperator.AND -> left.offersChoice() || right.offersChoice()
         }
 
+    override fun applyChoice(subtreeMatcher: SpdxExpression, choice: SpdxExpression): SpdxExpression {
+        if (!subtreeMatcher.isValidChoice(choice)) {
+            throw InvalidLicenseChoiceException(
+                "$choice is not a valid choice for the subtree expression. Valid choices are: ${validChoices()}."
+            )
+        }
+        if (subtreeMatcher.validChoices().size > 2) {
+            throw InvalidLicenseChoiceException(
+                "Subtree matcher $subtreeMatcher must not contain more than two choices."
+            )
+        }
+
+        return replaceMatcherWithChoice(subtreeMatcher, choice)
+    }
+
+    private fun replaceMatcherWithChoice(subtreeMatcher: SpdxExpression, choice: SpdxExpression) =
+        if (toString().contains(subtreeMatcher.toString())) {
+            toString().replace(subtreeMatcher.toString(), choice.toString()).toSpdx()
```

Comment:
Maybe avoid calling `toString()` on the same objects multiple times by introducing variables?